### PR TITLE
Second screen added

### PR DESCRIPTION
**Pull request by sroberts2**
Added the second screen to display a subset of the Company profile values.
Added initial error handling.
Added initial validation.
Added initial logging.

Helps Resolve: SFA-1260

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/lookup/controller/CompanyConfirmationController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.view.UrlBasedViewResolver;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+
+@Controller
+@RequestMapping("/accounts/confirmation")
```
Comment (drosser) [inline src/main/java/uk/gov/companieshouse/lookup/controller/CompanyConfirmationController.java]: Should this route be `/accounts/confirmation`? What if this was used outside of accounts?

File: src/main/java/uk/gov/companieshouse/lookup/controller/CompanyLookupController.java [hunk 2]
```diff
@@ -1,50 +1,79 @@
...
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
 import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
-import uk.gov.companieshouse.lookup.exception.ServiceException;
 import uk.gov.companieshouse.lookup.model.CompanyLookup;
 import uk.gov.companieshouse.lookup.service.CompanyLookupService;
+import uk.gov.companieshouse.lookup.validation.ValidationError;
+import uk.gov.companieshouse.lookup.validation.ValidationHandler;
 
 @Controller
 @RequestMapping("/accounts/lookup")
```
Comment (drosser) [inline src/main/java/uk/gov/companieshouse/lookup/controller/CompanyLookupController.java]: As above

File: src/main/java/uk/gov/companieshouse/lookup/exception/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -5,29 +5,29 @@
...
     public String handleRuntimeException(HttpServletRequest request, Exception ex) {
 
-        LOG.severe(ex.getMessage());
-//
+        LOG.errorRequest(request, ex.getMessage(), ex);
         return "error";
     }
 
-    @ExceptionHandler(value = { ServiceException.class })
+    @ExceptionHandler(value = {URIValidationException.class})
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/lookup/exception/GlobalExceptionHandler.java]: This is a checked exception so this handler should never be used

File: src/main/java/uk/gov/companieshouse/lookup/controller/CompanyLookupController.java [hunk 4]
```diff
@@ -1,50 +1,81 @@
...
+            attributes.addAttribute("forward", forward);
+            attributes.addFlashAttribute("companyConfirmation", companyConfirmation);
 
-        model.addAttribute("companyProfile", companyProfile);
+            return UrlBasedViewResolver.REDIRECT_URL_PREFIX +
+                FOUND_REDIRECT.expand(companyLookup.getCompanyNumber()).toString();
 
-        return UrlBasedViewResolver.REDIRECT_URL_PREFIX +
-            FOUND_REDIRECT.expand("").toString();
+        } catch (ApiErrorResponseException e) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/lookup/controller/CompanyLookupController.java]: Could this logic be handled within the service layer? Ideally the controller wouldn't care about constructing validation errors - that should be the job of the validator class.

File: src/main/java/uk/gov/companieshouse/lookup/service/impl/CompanyLookupServiceImpl.java [hunk 5]
```diff
@@ -17,25 +23,47 @@
...
+            error.setFieldPath("companyNumber");
+            error.setMessageKey("validation.number.invalid");
+            validationErrors.add(error);
         }
 
-        return companyProfileApi;
+        return validationErrors;
     }
 
+    private CompanyConfirmation mapCompany(CompanyProfileApi companyProfileApi) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/lookup/service/impl/CompanyLookupServiceImpl.java]: Could this logic exist within a transformer class?

**Pull request merged**